### DISCUSSION TITLE
PR for case insensitive search

### DISCUSSION THREAD

**Discussion by matejdro** [firefly-iii/firefly-iii#4775] [General]
As mentioned in the FAQ, Postgres is at fault for not allowing case insensitive search by default: https://docs.firefly-iii.org/firefly-iii/faq/other/#auto-complete-is-case-sensitive.

One workaround for this that I see is that we wrap query string in LOWER() function if postgresql database is used. Would you be interested in PR for this, at some point?

**Comment by JC5** [comment]
No, because that means Firefly III would have database specific query code, which was the point of ORM in the first place ;)

**Reply by matejdro** [reply]
> which was the point of ORM in the first place

I mean sure. But you can't excuse leaving dangling bugs in the app because of this. Maybe we could add lower() to all queries (presumably all databases support it?

**Reply by JC5** [reply]
I can, and I will.

Adding lower() to all queries won't solve the problem anyway, because some languages don't have lower/upper case or use different characters when going from upper to lower case, depending on the context.

**Reply by matejdro** [reply]
Right, but all of those different characters would not make any difference, because they are only used for search.

**Comment by janbuchar** [comment]
I encountered this problem with rules (like this guy https://github.com/firefly-iii/firefly-iii/issues/4018) and found out that the SQL `LIKE` operator is used for searching there. 

I understand why @JC5 doesn't want to introduce driver-specific code, but using `ILIKE` for Postgres would help a lot. Would you be open to this kind of special-casing? Would you use it if Eloquent ORM shipped it?
